### DISCUSSION TITLE
v7.5.0-preview.4 Release of PowerShell

### DISCUSSION THREAD

**Discussion by SeeminglyScience** [PowerShell/PowerShell#24222] [General]
## [7.5.0-preview.4] - 2024-08-28

**NOTE:** The Windows x86 and Linux ARM32 packages are not included with this release due to a known issue. The issue will be fixed in the next release.

### Engine Updates and Fixes

- RecommendedAction: Explicitly start and stop ANSI Error Color (#24065) (Thanks @JustinGrote!)
- Improve .NET overload definition of generic methods (#21326) (Thanks @jborean93!)
- Optimize the `+=` operation for a collection when it's an object array (#23901) (Thanks @jborean93!)
- Allow redirecting to a variable as experimental feature `PSRedirectToVariable` (#20381)

### General Cmdlet Updates and Fixes

- Change type of `LineNumber` to `ulong` in `Select-String` (#24075) (Thanks @Snowman-25!)
- Fix `Invoke-RestMethod` to allow `-PassThru` and `-Outfile` work together (#24086) (Thanks @jshigetomi!)
- Fix Hyper-V Remoting when the module is imported via implicit remoting (#24032) (Thanks @jborean93!)
- Add `ConvertTo-CliXml` and `ConvertFrom-CliXml` cmdlets (#21063) (Thanks @ArmaanMcleod!)
- Add `OutFile` property in `WebResponseObject` (#24047) (Thanks @jshigetomi!)
- Show filename in `Invoke-WebRequest -OutFile -Verbose` (#24041) (Thanks @jshigetomi!)
- `Set-Acl`: Do not fail on untranslatable SID (#21096) (Thanks @jborean93!)
- Fix the extent of the parser error when a number constant is invalid (#24024)
- Fix `Move-Item` to throw error when moving into itself (#24004)
- Fix up .NET method invocation with `Optional` argument (#21387) (Thanks @jborean93!)
- Fix progress calculation on `Remove-Item` (#23869) (Thanks @jborean93!)
- Fix WebCmdlets when `-Body` is specified but `ContentType` is not (#23952) (Thanks @CarloToso!)
- Enable `-NoRestart` to work with `Register-PSSessionConfiguration` (#23891)
- Add `IgnoreComments` and `AllowTrailingCommas` options to `Test-Json` cmdlet (#23817) (Thanks @ArmaanMcleod!)
- Get-Help may report parameters with `ValueFromRemainingArguments` attribute as pipeline-able (#23871)

### Code Cleanup

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@xtqqczze, @eltociear</p>

</summary>

<ul>
<li>Minor cleanup on local variable names within a method (#24105)</li>
<li>Remove explicit <code>IDE1005</code> suppressions (#21217) (Thanks @xtqqczze!)</li>
<li>Fix a typo in <code>WebRequestSession.cs</code> (#23963) (Thanks @eltociear!)</li>
</ul>

</details>

### Tools

- devcontainers: mount workspace in /PowerShell (#23857) (Thanks @rzippo!)

### Tests

- Add debugging to the MTU size test (#21463)

### Build and Packaging Improvements

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@bosesubham2011</p>

</summary>

<ul>
<li>Update third party notices (Internal 32128)</li>
<li>Update cgmanifest (#24163)</li>
<li>Fixes to Azure Public feed usage (#24149)</li>
<li>Add support for back porting PRs from GitHub or the Private Azure Repos (#20670)</li>
<li>Move to <code>9.0.0-preview.6.24327.7</code> (#24133)</li>
<li>update path (#24134)</li>
<li>Update to the latest NOTICES file (#24131)</li>
<li>Fix semver issue with updating cgmanifest (#24132)</li>
<li>Add ability to capture MSBuild Binary logs when restore fails (#24128)</li>
<li>add ability to skip windows stage (#24116)</li>
<li>chore: Refactor Nuget package source creation to use New-NugetPackageSource function (#24104)</li>
<li>Make Microsoft feeds the default (#24098)</li>
<li>Cleanup unused csproj (#23951)</li>
<li>Add script to update SDK version during release (#24034)</li>
<li>Enumerate over all signed zip packages (#24063)</li>
<li>Update metadata.json for PowerShell July releases (#24082)</li>
<li>Add macos signing for package files (#24015)</li>
<li>Update <code>install-powershell.sh</code> to support azure-linux (#23955) (Thanks @bosesubham2011!)</li>
<li>Skip build steps that do not have exe packages (#23945)</li>
<li>Update metadata.json for PowerShell June releases (#23973)</li>
<li>Create <code>powershell.config.json</code> for <code>PowerShell.Windows.x64</code> global tool (#23941)</li>
<li>Fix error in the vPack release, debug script that blocked release (#23904)</li>
<li>Add vPack release (#23898)</li>
<li>Fix exe signing with third party signing for WiX engine (#23878)</li>
<li>Update wix installation in CI (#23870)</li>
<li>Add checkout to fix TSA config paths (#23865)</li>
<li>Merge the <code>v7.5.0-preview.3</code> release branch to GitHub master branch</li>
<li>Update <code>metadata.json</code> for the <code>v7.5.0-preview.3</code> release (#23862)</li>
<li>Bump <code>PSResourceGet</code> to <code>1.1.0-preview1</code> (#24129)</li>
<li>Bump <code>github/codeql-action</code> from 3.25.8 to 3.26.0 (#23953) (#23999) (#24053) (#24069) (#24095) (#24118)</li>
<li>Bump <code>actions/upload-artifact</code> from 4.3.3 to 4.3.6 (#24019) (#24113) (#24119)</li>
<li>Bump <code>agrc/create-reminder-action</code> from 1.1.13 to 1.1.15 (#24029) (#24043)</li>
<li>Bump <code>agrc/reminder-action</code> from 1.0.12 to 1.0.14 (#24028) (#24042)</li>
<li>Bump <code>super-linter/super-linter</code> from 5.7.2 to 6.8.0 (#23809) (#23856) (#23894) (#24030) (#24103)</li>
<li>Bump <code>ossf/scorecard-action</code> from 2.3.1 to 2.4.0 (#23802) (#24096)</li>
<li>Bump <code>actions/dependency-review-action</code> from 4.3.2 to 4.3.4 (#23897) (#24046)</li>
<li>Bump <code>actions/checkout</code> from 4.1.5 to 4.1.7 (#23813) (#23947)</li>
<li>Bump <code>github/codeql-action</code> from 3.25.4 to 3.25.8 (#23801) (#23893)</li>
</ul>

</details>

### Documentation and Help Content

- Update docs sample nuget.config (#24109)
- Update Code of Conduct and Security Policy (#23811)
- Update working-group-definitions.md for the Security WG (#23884)
- Fix up broken links in Markdown files (#23863)
- Update Engine Working Group Members (#23803) (Thanks @kilasuit!)
- Remove outdated and contradictory information from `README` (#23812)

[7.5.0-preview.4]: https://github.com/PowerShell/PowerShell/compare/v7.5.0-preview.3...v7.5.0-preview.4

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - D8EA69CD72C01E80C2C84EE84BF9E8550B64583C49EF8A8C84E5F57ADFF68014
- powershell-7.5.0-preview.4-linux-arm32.tar.gz
  - 1B5C029CDC91974037E0FE2B04B80FE5206DAB55549A468340DDF77A4173AD09
- powershell-7.5.0-preview.4-linux-arm64.tar.gz
  - CEE68F97F3387740EBB18BE1EDDE0675E4135C2AA8C10BD57D8E8AAEEFED11E6
- powershell-7.5.0-preview.4-linux-musl-x64.tar.gz
  - 7448D91FE1E5DFA6239FA8717FA13EDAC3AB924AE7E8D1BBF88DD86E656EA527
- powershell-7.5.0-preview.4-linux-x64-fxdependent.tar.gz
  - 093A6B89A96A99B060FF979AD047E2E25CF0EA1BD5B35719B3CD3A5D05B72CE9
- powershell-7.5.0-preview.4-linux-x64-musl-noopt-fxdependent.tar.gz
  - 56658E0A121DD613CCCF9C02D34FABBECA7F470979DF8838E6B3F470AC779CF3
- powershell-7.5.0-preview.4-linux-x64.tar.gz
  - 66E12E9CDF219B505FC26A8098597D8AD5B53F303AA32222D3A2B72CC6729F41
- powershell-7.5.0-preview.4-osx-arm64.pkg
  - 294510DAB4E3F1F62C56C6DD2C2CE459E3077064484D461BB10613FEF35C9595
- powershell-7.5.0-preview.4-osx-arm64.tar.gz
  - 0E3E1E0F2F01A7D0F98F833B2D7F1F310DC94A2CACEA8D2ED4B8335259B573A6
- powershell-7.5.0-preview.4-osx-x64.pkg
  - B897C201F41E63430F40EA87A830EACDA9FF5F46A694D322DD0773042F604F6D
- powershell-7.5.0-preview.4-osx-x64.tar.gz
  - D13E8573FC08A48DAF797A475CAA6418784D345A86EA4937F0FBA6DE203DB336
- PowerShell-7.5.0-preview.4-win-arm64.msi
  - 3F21E7F67BA58A392C41E03EBB0141C532B370D6676041C314CCA45F82D57216
- PowerShell-7.5.0-preview.4-win-arm64.zip
  - FD122730A2102DEEB2216036E646379267CA64DF083EECC1C35919DD4982F1AD
- PowerShell-7.5.0-preview.4-win-fxdependent.zip
  - 71A0D8543E0A95B5392CB275ADED520A47EC8A3426DE92A71E74D577DB734C09
- PowerShell-7.5.0-preview.4-win-fxdependentWinDesktop.zip
  - 54CE3B5D82703F03C5DF98EE35441409FDEF6A300E23C9183BEB6D0ED03E4A89
- PowerShell-7.5.0-preview.4-win-x64.msi
  - C28D9A34EE6151EB35DD41A172626D2E91534DD3F5DEF31E329E733AC3F19F61
- PowerShell-7.5.0-preview.4-win-x64.zip
  - B4279D8E1920CEED04C762BB2A0B454365F4484AA55E257615DAA1FB0DEF02D5
- PowerShell-7.5.0-preview.4-win-x86.msi
  - E63541B8B659384FE43A8C0F3E99FDAB0DA9A94AE047EF1665D121FC03149229
- PowerShell-7.5.0-preview.4-win-x86.zip
  - 81B59A1C9C4863A740CBEFC46DD735A10FED8EA7A28003C79BBB17F57C29E36C
- powershell-preview_7.5.0-preview.4-1.deb_amd64.deb
  - A6625D28D37EFD50A41DF12AF60661E89B405510C3EF555C0972A7A47F7FCF87
- powershell-preview-7.5.0_preview.4-1.cm.aarch64.rpm
  - A5F1388DA9DA9C6FE02957F6986B9B1B7CFE739E0848B715BBCEE8A4816A06F6
- powershell-preview-7.5.0_preview.4-1.cm.x86_64.rpm
  - 82D7754BE69D3CA44EF59EE11D15E2975AB56832688FDE03F62D6D30F22E4323
- powershell-preview-7.5.0_preview.4-1.rh.x86_64.rpm
  - 0B40A1CFED3080030718C0FDF7074A541B86A9544A0B44364456230B25695231

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.5.0-preview.4'>v7.5.0-preview.4 Release of PowerShell</a>.</em>

**Comment by peetrike** [comment]
There is no x86 version for Windows.  Is this intended?

**Reply by jborean93** [reply]
> NOTE: The Windows x86 and Linux ARM32 packages are not included with this release due to a known issue. The issue will be fixed in the next release.

There was an issue with the .NET preview 6 on x86 that was causing pwsh to crash. It has been fixed in .NET 9 preview 7 and thus the next pwsh preview.

**Comment by peetrike** [comment]
and when using x64 MSI-based install, the install process never ends.  The msiexec process starts pwsh.exe process that doesn't respond.  When I kill that process, the setup finishes, but it seems, that the install is broken (PowerShell never shows prompt).

**Reply by JustinGrote** [reply]
x64 msi worked for me on Win11

**Reply by peetrike** [reply]
Win10 here.  It might happen, that corporate security agents interfere with setup, but Preview 3 installs without problems ...

**Reply by kellyelton** [reply]
This happened to me as well. I force killed a background powershell process that the installer started, and then the installer finally moved forward, but said it failed.
I then ran the installer and ran a 'Repair' which succeeded.
Now when I try to run powershell preview, it opens the console host and never shows any text, it just hangs forever. It also does the same thing inside of vscode.
I tried uninstalling, rebooting, and reinstalling, and have the same issue.
Windows 11 23H2 build 22631.4169

**Reply by kellyelton** [reply]
Enabling any of the highlighted settings will cause the installer to hang.

![image](https://github.com/user-attachments/assets/c1f710d2-b848-4395-8500-fccd83b38703)
![image](https://github.com/user-attachments/assets/6969040d-d0b7-412e-96c1-26f3f34d65f0)

**Reply by kellyelton** [reply]
v7.5.0-preview.3 installs as expected on my machine.

**Comment by nasasky39** [comment]
x64 msi worked for me on Win11 24H2

![WIN](https://github.com/user-attachments/assets/4f4c107b-4e57-4f28-87c0-3e2d182dd11b)
![WIN2](https://github.com/user-attachments/assets/30135f23-dca0-4f8a-8a4c-4a1d4a122550)

**Comment by kellyelton** [comment]
Powershell just hangs when trying to open it in Windows Terminal or Visual Studio Code on Windows 11.
I've tried repairing, uninstalling, rebooting, and reinstalling.

**Reply by kellyelton** [reply]
v7.5.0-preview.3 works as expected on my machine.
